project/npda/views/patient.py

```diff
@@ -45,37 +47,37 @@ def get_queryset(self):
         """
         pz_code = self.request.session.get("sibling_organisations").get("pz_code")
         ods_code = self.request.session.get("ods_code")
+        filtered_patients = None
         # filter patients to the view preference of the user
         if self.request.user.view_preference == 0:
             # organisation view
-            return (
-                Patient.objects.filter(site__organisation_ods_code=ods_code)
-                .annotate(
-                    visit_error_count=Count(Case(When(visit__is_valid=False, then=1)))
-                )
-                .order_by("is_valid", "visit_error_count", "pk")
+            filtered_patients = Q(
+                site__organisation_ods_code=ods_code,
             )
         elif self.request.user.view_preference == 1:
             # PDU view
-            return (
-                Patient.objects.filter(site__paediatric_diabetes_unit_pz_code=pz_code)
-                .annotate(
-                    visit_error_count=Count(Case(When(visit__is_valid=False, then=1)))
-                )
-                .order_by("is_valid", "visit_error_count", "pk")
-            )
+            filtered_patients = Q(site__paediatric_diabetes_unit_pz_code=pz_code)
         elif self.request.user.view_preference == 2:
-            # National view
-            return (
-                Patient.objects.all()
-                .annotate(
-                    visit_error_count=Count(Case(When(visit__is_valid=False, then=1)))
-                )
-                .order_by("is_valid", "visit_error_count", "pk")
-            )
+            # National view - no filter
+            pass
         else:
             raise ValueError("Invalid view preference")
 
+        patient_queryset = Patient.objects.filter(
+            audit_cohorts__submission_active=True,
+        )
+        if filtered_patients is not None:
+            patient_queryset = patient_queryset.filter(filtered_patients)
+
+        AuditCohort = apps.get_model(app_label="npda", model_name="AuditCohort")
```

Comment:
Does this instance of AuditCohort need to be defined here? I can't see it referenced anywhere else

Reply:
if not used agree, do remove.